File: Exercism Codes/Python/list_ops.py
def append(list1, list2):
    return list1 + list2


def concat(lists):
    result = []
    for lst in lists:
        result += lst

    return result


def filter(function, list):
    return [x for x in list if function(x)]


def length(list):
    return len(list)


def map(function, list):
    return [function(x) for x in list]


def foldl(function, list, initial):
    for item in list:
        initial = function(initial, item)

    return initial


def foldr(function, list, initial):
    for item in reverse(list):
        initial = function(initial, item)

    return initial


def reverse(list):
    return list[::-1]

File: Exercism Codes/JavaScript/list-ops.js
export class List {
  constructor(values = []) {
    this.values = values;
  }

  append(other) {
    return new List(this.values.concat(other.values));
  }

  concat(other) {
    let newValues = this.values;
    for (const x of other.values) {
      newValues = newValues.concat(x.values);
    }
    return new List(newValues);
  }

  filter(func) {
    return new List(this.values.filter(func));
  }

  map(func) {
    return new List(this.values.map(func));
  }

  length() {
    return this.values.length;
  }

  foldl(func, initial) {
    for (let i = 0; i < this.length(); i++) {
      initial = func(initial, this.values[i]);
    }
    return initial;
  }

  foldr(func, initial) {
    for (let i = this.length() - 1; i >= 0; i--) {
      initial = func(initial, this.values[i]);
    }
    return initial;
  }

  reverse() {
    return new List(this.values.reverse());
  }
}